src/httpfetch.cpp

```diff
@@ -393,10 +385,16 @@ const HTTPFetchResult * HTTPFetchOngoing::complete(CURLcode res)
 	}
 
 	if (res != CURLE_OK) {
-		errorstream << request.url << " not found ("
-			<< curl_easy_strerror(res) << ")"
-			<< " (response code " << result.response_code << ")"
+		errorstream << "HTTPFetch for " << request.url << " failed ("
+			<< curl_easy_strerror(res) << ")" << std::endl;
+	} else if (result.response_code >= 400) {
+		errorstream << "HTTPFetch for " << request.url
+			<< " returned response code " << result.response_code
 			<< std::endl;
+		if (!result.data.empty()) {
+			errorstream << "Response body:" << std::endl
```

Comment:
I don't think it's a good idea to dump the entire body for _all_ failing requests (remember mods can invoke these too).
Shouldn't the serverlist stuff have a callback where it could do this?

Reply:
Updated so that only the server list prints the body.